Added a select function (issue #38)

bhazelton opened the pull request:
Includes options for selecting antennas (by name or by number), frequencies, times, polarizations and even specific baseline-times (the latter probably won't be widely used but is needed for some kinds of file matching desired in the future, see issue #111).

The function is fully covered by tests, has a doc string, and I've added some tutorial examples on selecting, although the examples don't cover all the possibilities.

There are warnings (and associated errors in the appropriate write functions) for selections of frequencies or polarizations that will break miriad or uvfits file formats. The errors in the write functions are also covered by the tests.

adampbeardsley commented:
I just made a few commits to round out the select and tests a little bit.
My only remaining concern is whether we want to use random functions in the tests -- they're pretty simple here, but I worry it may make reproducibility difficult.

bhazelton commented:
Those improvements look good, Adam, thank you!

bhazelton commented:
I started with random tests partly to make sure I didn't miss anything, but I agree with your concern about reproducibility, Adam. So I converted the tests not to use a random choice. I also cleaned up the multiple file reads in the tests by copying the objects.
